lib/src/platform_checkbox.dart

```diff
@@ -0,0 +1,184 @@
+    this.value,
+    this.tristate = false,
+    required this.onChanged,
+    this.activeColor,
+    this.checkColor,
+    this.focusColor,
+    this.focusNode,
+    this.autofocus = false,
+    this.shape,
+    this.side,
+  }) : assert(tristate || value != null);
+  final Key? widgetKey;
+  final bool? value;
+  final bool tristate;
+  final ValueChanged<bool?>? onChanged;
+  final Color? activeColor;
+  final Color? checkColor;
+  final Color? focusColor;
+  final FocusNode? focusNode;
+  final bool autofocus;
+  final OutlinedBorder? shape;
+  final BorderSide? side;
+}
+
+class MaterialCheckboxData extends _BaseData {
+  MaterialCheckboxData({
+    // Common
+    super.widgetKey,
+    super.value,
+    super.tristate = false,
+    required super.onChanged,
+    super.activeColor,
+    super.checkColor,
+    super.focusColor,
+    super.focusNode,
+    super.autofocus = false,
+    super.shape,
+    super.side,
+    //Material
+    this.mouseCursor,
+    this.fillColor,
+    this.hoverColor,
+    this.overlayColor,
+    this.splashRadius,
+    this.materialTapTargetSize,
+    this.visualDensity,
+    this.isError = false,
+  });
+
+  final MouseCursor? mouseCursor;
+  final MaterialStateProperty<Color?>? fillColor;
+  final Color? hoverColor;
+  final MaterialStateProperty<Color?>? overlayColor;
+  final double? splashRadius;
+  final MaterialTapTargetSize? materialTapTargetSize;
+  final VisualDensity? visualDensity;
+  final bool isError;
+}
+
+class CupertinoCheckboxData extends _BaseData {
+  CupertinoCheckboxData({
+    //Common
+    super.widgetKey,
+    super.value,
+    super.tristate = false,
+    required super.onChanged,
+    super.activeColor,
+    super.checkColor,
+    super.focusColor,
+    super.focusNode,
+    super.autofocus = false,
+    super.shape,
+    super.side,
+
+    //Cupertino
+    this.inactiveColor,
+  });
+
+  final Color? inactiveColor;
+}
+
+class PlatformCheckbox extends PlatformWidgetBase<CupertinoCheckbox, Checkbox> {
+  //Common
+  final Key? widgetKey;
+  final bool? value;
+  final bool tristate;
+  final ValueChanged<bool?>? onChanged;
+  final Color? activeColor;
+  final Color? checkColor;
+  final Color? focusColor;
+  final FocusNode? focusNode;
+  final bool autofocus;
+  final OutlinedBorder? shape;
+  final BorderSide? side;
+
+  //Platform
+  final PlatformBuilder<MaterialCheckboxData>? material;
+  final PlatformBuilder<CupertinoCheckboxData>? cupertino;
+
+  PlatformCheckbox({
+    //Common
+    super.key,
+    this.widgetKey,
+    this.value,
+    this.tristate = false,
+    required this.onChanged,
+    this.activeColor,
+    this.checkColor,
+    this.focusColor,
+    this.focusNode,
+    this.autofocus = false,
+    this.shape,
+    this.side,
+    //Platform
+    this.material,
+    this.cupertino,
+  }) : assert(tristate || value != null);
+
+  @override
+  Checkbox createMaterialWidget(BuildContext context) {
+    final data = material?.call(context, platform(context));
```

Comment:
Need to put the assert here that combines the properties as a result from the widget props and the data props.

```
final tristate = data?.tristate ?? this.tristate;
final value = data?.value ?? this.value;
assert(tristate || value != null);

```